src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurRequestExtractFileValidationServiceImpl.java

```diff
@@ -0,0 +1,46 @@
+package edu.cornell.kfs.concur.batch.service.impl;
+
+import java.util.List;
+
+import edu.cornell.kfs.concur.batch.businessobject.ConcurRequestExtractFile;
+import edu.cornell.kfs.concur.batch.businessobject.ConcurRequestExtractRequestDetailFileLine;
+import edu.cornell.kfs.concur.batch.service.ConcurRequestExtractFileValidationService;
+
+import org.kuali.rice.core.api.util.type.KualiDecimal;
+
+public class ConcurRequestExtractFileValidationServiceImpl implements ConcurRequestExtractFileValidationService {
+
+    private static org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(ConcurRequestExtractFileValidationServiceImpl.class);
+
+    public boolean fileRowCountMatchesHeaderRowCount(ConcurRequestExtractFile requestExtractFile) {
+        int countEntryDetailLines = 0;
+        List<ConcurRequestExtractRequestDetailFileLine> requestDetailLines = requestExtractFile.getRequestDetails();
```

Comment:
Technically this attribute should not be null as it is initialized in the class constructor but that condition will be taken into account when requested new method is created.